### PR TITLE
refactor: proposal token channel

### PR DESCRIPTION
**Pull request by tuliomir**
### Acceptance Criteria
- The fetching of proposal token metadata should happen within a channel with controlled concurrency queue

### Security Checklist
- [X] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: src/sagas/tokens.js [hunk 1]
```diff
@@ -281,6 +283,48 @@ export function* monitorSelectedToken() {
...
+ * This saga will create a channel to queue PROPOSAL_TOKEN_FETCH_REQUESTED actions and
+ * consumers that will run in parallel consuming those actions.
+ *
+ * More information about channels can be read on https://redux-saga.js.org/docs/api/#takechannel
+ */
+function* fetchProposalTokenDataQueue() {
+  const fetchProposalTokenDataChannel = yield call(channel);
+
+  // Fork CONCURRENT_FETCH_REQUESTS threads to download token balances
+  for (let i = 0; i < METADATA_CONCURRENT_DOWNLOAD; i += 1) {
```
Comment (pedroferreira1) [inline src/sagas/tokens.js:296]: I think we should use a smaller quantity here. The metadata is requested from the explorer service, the token data is requested from the full node. Why don't we start with 3 or 5? It should be hard to have any tx with more than this amount of different tokens.

Comment (tuliomir) [reply src/sagas/tokens.js]: Good point, on [a07b2cc](https://github.com/HathorNetwork/hathor-wallet/pull/401/commits/a07b2cc965d57036cdad292ed80b63884c8f31a5) changed it to the `CONCURRENT_FETCH_REQUESTS` constant already used for fetching token metadata from the fullnode. Currently its value is `5`.

File: src/sagas/tokens.js [hunk 2]
```diff
@@ -281,6 +283,48 @@ export function* monitorSelectedToken() {
...
+    const action = yield take(types.PROPOSAL_TOKEN_FETCH_REQUESTED);
+    yield put(fetchProposalTokenDataChannel, action);
+  }
+}
+
+/**
+ * This saga will consume the fetchProposalTokenDataChannel for PROPOSAL_TOKEN_FETCH_REQUESTED actions
+ * and wait until the PROPOSAL_TOKEN_FETCH_SUCCESS action is dispatched with the specific proposalId
+ */
+function* fetchProposalTokenDataConsumer(fetchTokenMetadataChannel) {
```
Comment (pedroferreira1) [inline src/sagas/tokens.js:310]: This is not a token metadata channel, right? It's more like a token data

File: src/sagas/tokens.js [hunk 3]
```diff
@@ -281,6 +283,48 @@ export function* monitorSelectedToken() {
...
+/**
+ * This saga will create a channel to queue PROPOSAL_TOKEN_FETCH_REQUESTED actions and
+ * consumers that will run in parallel consuming those actions.
+ *
+ * More information about channels can be read on https://redux-saga.js.org/docs/api/#takechannel
+ */
+function* fetchProposalTokenDataQueue() {
+  const fetchProposalTokenDataChannel = yield call(channel);
+
+  // Fork CONCURRENT_FETCH_REQUESTS threads to download token metadata
```
Comment (pedroferreira1) [inline src/sagas/tokens.js:295]: Is it token metadata or token data?

File: src/sagas/tokens.js (hunk 2 above)
Comment (tuliomir) [reply src/sagas/tokens.js]: On [ad3eace](https://github.com/HathorNetwork/hathor-wallet/pull/401/commits/ad3eace4c78441d95903f2e43d881ca63d70d740) adjusted jsdocs, variable names and error messages to clarify that this context is about token data ( name and symbol )

File: src/sagas/tokens.js (hunk 3 above)
Comment (tuliomir) [reply src/sagas/tokens.js]: On [ad3eace](https://github.com/HathorNetwork/hathor-wallet/pull/401/commits/ad3eace4c78441d95903f2e43d881ca63d70d740) adjusted jsdocs, variable names and error messages to clarify that this context is about token data ( name and symbol )

**Review by pedroferreira1 — APPROVED** [review]

**Review by r4mmer — APPROVED** [review]

**Pull request merged**